### PR TITLE
Feature/options menu

### PR DESCRIPTION
**Pull request by TArchetype**
Added an functional options menu

Performed a general face lift of all UI elements in the menu.
Added a simple preset color scheme.
Added a tab system

### CODE REVIEW

**Comment by toberge** [comment]
I don't think the options menu is supposed to show up instead of the landing menu?

**Comment by toberge** [comment]
Other issues:

- All sliders have the deselect-when-selected problem
- The LB/RB labels don't show what you're supposed to press on the keyboard
- Whatever this is:
image

Other remarks:

- Nice font
- Nice colors

File: Assets/Scripts/Gamestate/PlayerIdentity.cs.orig
Comment (toberge) [reply Assets/Scripts/Gamestate/PlayerIdentity.cs.orig:22]: oh. um. Please don't leave backup files like this one and the Menu scene backup, Git takes care of that :wink:

File: Assets/ScriptableObjects/Input/PlayerControls.inputactions
Comment (toberge) [inline Assets/ScriptableObjects/Input/PlayerControls.inputactions:54]: Perhaps these should be called something else than `Left/RightButton`, this is what we could call the `A/D` keybinds that we already have (though in a different way).

File: Assets/Scripts/UI/TabsButton.cs
Comment (toberge) [inline Assets/Scripts/UI/TabsButton.cs:6]: TabButton without the **s**, please.

**Review by toberge — CHANGES_REQUESTED** [review]
Issues:

- Goes straight to options menu :(
- All sliders have the deselect-when-selected problem
- You broke Fredrik's smooth animation for when players join >:(
- The LB/RB labels don't show what you're supposed to press on the keyboard
- ~~Whatever this is:~~ sorry, that was Unity bugging the heck out on my end :upside_down_face: 
image
- Also this:
image

Other remarks:

- Nice font
- Nice colors (I like that we got red and yellow as player colors, feels Martian)
- Nice set of settings
- Nice tabs
- The other menus look a lot better as well

File: Assets/Scripts/UI/TabGroup.cs
Comment (TArchetype) [reply Assets/Scripts/UI/TabGroup.cs]: fixed

File: Assets/Scripts/UI/TabGroup.cs
Comment (TArchetype) [reply Assets/Scripts/UI/TabGroup.cs]: fixed

File: Assets/Scripts/UI/TabsButton.cs
Comment (TArchetype) [reply Assets/Scripts/UI/TabsButton.cs:6]: This is a conflict issue, "TabButton" is used by Unity's ShaderGraph. Potentially renamed to "TabButtonController"?

File: Assets/ScriptableObjects/Input/PlayerControls.inputactions
Comment (TArchetype) [reply Assets/ScriptableObjects/Input/PlayerControls.inputactions]: Renamed to "LeftTab" and "RightTab"

File: Assets/Scripts/Gamestate/PlayerIdentity.cs.orig
Comment (TArchetype) [reply Assets/Scripts/Gamestate/PlayerIdentity.cs.orig]: Should be fixed now? Not quite sure how this happend.

**Review by Fueredoriku — CHANGES_REQUESTED** [review]
Looks amazing!! 😮

I left a couple of comments that should be adressed (Even though they are pretty much non-functional critque)

Visually my only gripe is that the "back"-buttons in the tabs aren't styled the same:

image
vs.
image

 Everything else looks very good! 😎👍
 Read the code and tested.

File: Assets/Scripts/Control&Input/InputManager.cs
Comment (TArchetype) [reply Assets/Scripts/Control&Input/InputManager.cs:20]: I agree with the generalization to an action rather than specific button, I've renamed the variables to `onLeftTabPerformed` and `OnRightTabPerformed`, and similarly reflected the change in the inputMap.

**Review by Fueredoriku — APPROVED** [review]
Looks good!
Seems like the back-buttons mentioned in the previews review still are different, but it's not that important:
I'll approve and then you can decide if we should have it like is and possibly change later, or alternatively change before merge.
Either way: Good work! 👍

File: Assets/Scripts/Gamestate/PlayerIdentity.cs.orig
Comment (toberge) [inline Assets/Scripts/Gamestate/PlayerIdentity.cs.orig]: Nice that you removed the merge markers, though you should still **remove this file**.

File: Assets/Scripts/UI/SliderFocus.cs
Comment (toberge) [inline Assets/Scripts/UI/SliderFocus.cs:7]: Nice solution to the slider issue!

**Review by toberge — CHANGES_REQUESTED** [review]
Looks even better now!

I still have some remarks, and I'd like you to fix the discrepancies in layout between the two settings tabs. Audio seems to have larger outer margins for the text and controls than the graphics tab, and the back buttons are still in different positions. Please set the same padding in the vertical layout groups, use stretch for the audio tab as well, and use the same layout approach as either one of the back buttons (I'd prefer the approach used in the Audio tab, but adjust it as you'd like).

Sorry about these extensive UI complaints, but I'm doing way too much frontend development at work :sweat_smile:

File: Assets/Scripts/UI/TabsButton.cs
Comment (toberge) [reply Assets/Scripts/UI/TabsButton.cs:6]: Ouch. Well, that's fine then.

File: Assets/Scripts/Gamestate/PlayerIdentity.cs.orig
Comment (toberge) [reply Assets/Scripts/Gamestate/PlayerIdentity.cs.orig]: This goes for the .orig version of Menu as well. Git takes care of previous versions. You could try to checkout versions of a file from the last commit instead of creating a bunch of these superfluous files.

**Review by Fueredoriku — CHANGES_REQUESTED** [review]
I just realised that the feedback from @toberge has been marked as resolved without being resolved. (mainly deletion of .orig files and merge headers 😬) This needs to be fixed before merge

File: Assets/Scripts/Gamestate/PlayerIdentity.cs.orig
Comment (toberge) [reply Assets/Scripts/Gamestate/PlayerIdentity.cs.orig]: Yeah nope, see the corresponding comment in my new review

Comment (toberge) [reply Assets/Scripts/Gamestate/PlayerIdentity.cs.orig]: And please don't resolve a thread when you're not sure if it has been fixed :slightly_smiling_face:

Comment (TArchetype) [reply Assets/Scripts/Gamestate/PlayerIdentity.cs.orig]: Thought 'resolve' was referring to the review itself, but the more you learn. :) The backup files are now removed.

File: Assets/Scripts/UI/SliderFocus.cs
Comment (TArchetype) [reply Assets/Scripts/UI/SliderFocus.cs:7]: I don't mind any of the comments, I'm still rather new to any kind of UI development, so every nitpick helps 👍. Most likely we'll do an entire new redesign of the UI once we have more models, textures, etc. available and a theme locked down for the game.

**Comment by Fueredoriku** [comment]
image

Currently experiencing this, I'll make a commit to fix it

**Review by Fueredoriku — APPROVED** [review]
Looks good now! 
Seems there was an issue (probably stemming from rebase) with index out of bounds exception for inputs.
I fixed it in 52df40d8ef10cfbc80ecf99d2313ff5e9e815f44 , so I approve with the inclusion of that commit 👍

File: Assets/Scripts/Control&Input/PlayerInputManagerController.cs
Comment (toberge) [inline Assets/Scripts/Control&Input/PlayerInputManagerController.cs:54]: @Fueredoriku doesn't the indexing issue affect this as well?

File: Assets/Scripts/UI/SliderFocus.cs
Comment (toberge) [reply Assets/Scripts/UI/SliderFocus.cs:7]: Yeah. It's nice to have a base that doesn't have any glaring flaws. I suspect we will keep the settings menu quite close to its current layout though, since this is fairly standard.

File: Assets/Scripts/Control&Input/PlayerInputManagerController.cs
Comment (Fueredoriku) [reply Assets/Scripts/Control&Input/PlayerInputManagerController.cs:54]: One would think so, but my testing proved otherwise

Comment (toberge) [reply Assets/Scripts/Control&Input/PlayerInputManagerController.cs]: Haha I was right. Fixed it along with some other issues.

**Review by toberge — APPROVED** [review]
With the changes I just made, I approve. I'll leave it to you @TArchetype to accept the adjustment I made to the graphics options layout, I just made the text stay in the same spot as in the audio tab, though I did so by increasing the height of the entire items, so please have a look and adjust them again if you'd like. Should be possible to adjust the height of the dropdowns individually if you disable the height constraints on the field containers.

**Comment by TArchetype** [comment]
> With the changes I just made, I approve. I'll leave it to you @TArchetype to accept the adjustment I made to the graphics options layout, I just made the text stay in the same spot as in the audio tab, though I did so by increasing the height of the entire items, so please have a look and adjust them again if you'd like. Should be possible to adjust the height of the dropdowns individually if you disable the height constraints on the field containers.

I liked the change 👍

**Pull request merged**
